### PR TITLE
Download Android AOT tools to the artifacts cache and build APKs usin…

### PR DESCRIPTION
**Pull request by jason-simmons**
…g these tools

Also update the engine to a build where these tools have been uplaoded

### CODE REVIEW

File: packages/flutter_tools/lib/src/build_configuration.dart [hunk 1]
```diff
@@ -24,6 +24,10 @@ enum BuildMode {
 
 String getModeName(BuildMode mode) => getEnumName(mode);
 
+// Returns true if the selected build mode uses ahead-of-time compilation.
+bool isAotBuildMode(BuildMode mode) =>
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/build_configuration.dart]: I think the style guide only wants `=>` if it fits on one line...

File: packages/flutter_tools/lib/src/cache.dart [hunk 2]
```diff
@@ -190,6 +190,7 @@ class FlutterEngine {
   List<String> _getEngineDirs() {
     List<String> dirs = <String>[
       'android-arm',
+      'android-arm-profile',
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/cache.dart]: 👍

File: packages/flutter_tools/lib/src/cache.dart [hunk 3]
```diff
@@ -202,13 +203,17 @@ class FlutterEngine {
     return dirs;
   }
 
-  List<String> _getToolsDirs() {
+  List<List<String>> _getToolsDirs() {
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/cache.dart]: Could you add a brief doc to this? It looks like it returns a list of tuples of target dir paths and source google storage paths?

**Comment by devoncarew** [comment]
Does this now work for non-engine builds? Perhaps remove the `(local engine builds only)` text?

File: packages/flutter_tools/lib/src/toolchain.dart [hunk 4]
```diff
@@ -163,13 +163,13 @@ class ToolConfiguration {
 
       // Return something like 'out/android_Release'.
       String buildOutputPath = 'out/${type}_$_modeStr';
-      if (mode == BuildMode.profile)
+      if (isAotBuildMode(mode))
         buildOutputPath += '_Deploy';
 
       return new Directory(path.join(engineSrcPath, buildOutputPath));
     } else {
       // For now, only suffix for deploy variants.
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/toolchain.dart]: We should update this comment.

**Comment by devoncarew** [comment]
lgtm!

File: packages/flutter_tools/lib/src/build_configuration.dart (hunk 1 above)
Comment (jason-simmons) [inline packages/flutter_tools/lib/src/build_configuration.dart]: done

File: packages/flutter_tools/lib/src/cache.dart (hunk 3 above)
Comment (jason-simmons) [inline packages/flutter_tools/lib/src/cache.dart]: done

File: packages/flutter_tools/lib/src/toolchain.dart (hunk 4 above)
Comment (jason-simmons) [inline packages/flutter_tools/lib/src/toolchain.dart]: done

**Pull request merged**